src/Orleans/Core/IInternalGrainFactory.cs

```diff
@@ -47,5 +47,21 @@ TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId, SiloAddress de
         /// A reference to <paramref name="grain"/> which implements <paramref name="interfaceType"/>.
         /// </returns>
         object Cast(IAddressable grain, Type interfaceType);
+
+        /// <summary>
+        /// Gets a reference to the grain with the provided id.
+        /// </summary>
+        /// <typeparam name="TGrainInterface">The grain reference interface type.</typeparam>
+        /// <param name="grainId">The grain id.</param>
+        /// <returns>A reference to the grain with the provided id.</returns>
+        TGrainInterface GetGrain<TGrainInterface>(GrainId grainId);
```

Comment:
it's OK... you are right, this is an overload of other methods with the same dilemma. Let's keep as is.